src/frontends/jax/src/utils.cpp

```diff
@@ -0,0 +1,177 @@
+    ov::Output<ov::Node> tmp_lhs = lhs;
+    ov::Output<ov::Node> tmp_rhs = rhs;
+    // Python scalar has lower priority than any tensor with any dimension.
+    // If only one input is PyScalar, replace it with const to mitigate issues with dynamic type caused by dynamic
+    // shape.
+    if (is_lhs_python_scalar && !is_rhs_python_scalar) {
+        tmp_lhs = context.mark_node(std::make_shared<opset10::ConvertLike>(const_0, lhs));
+        tmp_rhs = context.mark_node(std::make_shared<opset10::ConvertLike>(const_1, rhs));
+    } else if (!is_lhs_python_scalar && is_rhs_python_scalar) {
+        tmp_lhs = context.mark_node(std::make_shared<opset10::ConvertLike>(const_1, lhs));
+        tmp_rhs = context.mark_node(std::make_shared<opset10::ConvertLike>(const_0, rhs));
+    }
+
+    auto at = context.mark_node(
+        std::make_shared<ov::op::v14::ConvertPromoteTypes>(tmp_lhs, tmp_rhs, true, true, element::f32));
+    auto dst_type = at->get_output_element_type(0);
+    if (dst_type.is_dynamic()) {
+        // Add ConvertLike on original node to not remove changes to shape done to differentiate between tensors and
+        // scalars.
+        lhs = context.mark_node(std::make_shared<opset10::ConvertLike>(lhs, at->output(0)));
+        rhs = context.mark_node(std::make_shared<opset10::ConvertLike>(rhs, at->output(1)));
+    } else {
+        // Cast to destination type
+        if (dst_type != lhs_type) {
+            lhs = context.mark_node(std::make_shared<opset10::Convert>(lhs, dst_type));
+        }
+        if (dst_type != rhs_type) {
+            rhs = context.mark_node(std::make_shared<opset10::Convert>(rhs, dst_type));
+        }
+    }
+    return;
+}
+
+void align_output_types(const NodeContext& context, OutputVector& outputs) {
+    for (size_t i = 0; i < outputs.size(); i++) {
+        auto dtype_any = context.get_output_type(i);
+        if (dtype_any.is<element::Type>()) {
+            auto dtype = dtype_any.as<element::Type>();
+            if (dtype.is_static() && dtype != outputs[i].get_element_type()) {
+                outputs[i] = std::make_shared<opset10::Convert>(outputs[i], dtype);
+            }
+        }
+    }
+}
+
+Output<Node> get_input_with_floating_type(const NodeContext& context, size_t idx) {
+    auto x = context.get_input(static_cast<int>(idx));
+    // This const only needed for type alignment
+    auto dummy_const = context.mark_node(ov::op::v0::Constant::create(element::f32, Shape({}), {0.5}))->output(0);
+    align_eltwise_input_types(context, x, dummy_const, false, true);
+    return x;
+}
+
+Output<Node> get_input_as_i32(const NodeContext& context, size_t idx) {
+    auto x = context.get_input(static_cast<int>(idx));
+    if (x.get_element_type() != element::i32) {
+        x = context.mark_node(std::make_shared<ov::op::v0::Convert>(x, element::i32));
+    }
+    return x;
+}
+
+std::tuple<Output<Node>, Output<Node>> get_inputs_with_promoted_types(const NodeContext& context,
```

Comment:
Is this needed for Jax?

Reply:
When two inputs with different types are given to an operation, JAX has similar behaviors with PyTorch, that takes the type with higher precision/range as the output type. If I'm not misunderstanding it, I think we need this API for JAX.